src/window.cpp

```diff
@@ -93,10 +93,14 @@ bool _mouse_hovering;      ///< The mouse is hovering over the same point.
 SpecialMouseMode _special_mouse_mode; ///< Mode of the mouse.
 
 /**
- * List of all WindowDescs.
- * This is a pointer to ensure initialisation order with the various static WindowDesc instances.
+ * Get list of all WindowDescs.
+ * @return List of all WindowDescs.
  */
-std::vector<WindowDesc*> *_window_descs = nullptr;
+std::vector<WindowDesc *> &GetWindowDescs()
+{
+	static std::vector<WindowDesc *> *s_window_descs = new std::vector<WindowDesc *>();
```

Comment:
Why do we allocate the vector at all? Can't it be just `static std::vector<WindowDesc *> s_window_descs;`?
Or is that due to the SIOF thing? But static local variables have a pretty well-defined behaviour, so I would reckon it is not be affected by SIOF.

Reply:
That fixes SIOF, but as far as I know not de-initialisation order.

Reply:
Maybe worth adding a comment to clarify that's why it's a pointer?